archive/tar.go

```diff
@@ -80,6 +81,10 @@ func WriteDiff(ctx context.Context, w io.Writer, a, b string, opts ...WriteDiffO
 			return fmt.Errorf("failed to apply option: %w", err)
 		}
 	}
+	if tm := epoch.FromContext(ctx); tm != nil && options.SourceDateEpoch == nil {
```

Comment:
Like the gRPC header discussion above, should we use context for passing parameters like SourceDateEpoch?

Reply:
Yes, it is hard to specify the epoch consistently without the ctx